platform.go

```diff
@@ -92,18 +95,38 @@ var (
 // SupportedPlatforms returns the full list of supported platforms for
 // the version of Go that is
 func SupportedPlatforms(v string) []Platform {
-	if strings.HasPrefix(v, "go1.0") {
-		return Platforms_1_0
-	} else if strings.HasPrefix(v, "go1.1") {
-		return Platforms_1_1
-	} else if strings.HasPrefix(v, "go1.3") {
-		return Platforms_1_3
-	} else if strings.HasPrefix(v, "go1.4") {
-		return Platforms_1_4
-	} else if strings.HasPrefix(v, "go1.5") {
+	// go-version only cares about version numbers
+	ver := strings.Replace(v, "go", "", 1)
+
+	current, err := version.NewVersion(ver)
+	if err != nil {
+		log.Printf("Unable to parse current go version: %s\n%s", ver, err.Error())
```

Comment:
yeah we use it in [go.go](https://github.com/mitchellh/gox/blob/50c78f896d74353ff09c5b19e7e09a09c0469d12/go.go#L143), which is the only reason I've used it instead of failing silently.